2-servers/src/main/scala/com/snowplowanalytics/piinguin/server/clients/DynamoDBClient.scala

```diff
@@ -0,0 +1,149 @@
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+
+package com.snowplowanalytics.piinguin.server.clients
+
+// Scanamo
+import com.gu.scanamo._
+import com.gu.scanamo.error.DynamoReadError
+import com.gu.scanamo.syntax._
+import com.gu.scanamo.ops.ScanamoOps
+
+// DynamoDb
+import com.amazonaws.services.dynamodbv2.{
+  AmazonDynamoDBAsync,
+  AmazonDynamoDBAsyncClient,
+  AmazonDynamoDBAsyncClientBuilder
+}
+import com.amazonaws.auth.{AWSStaticCredentialsProvider, BasicAWSCredentials, AWSCredentials}
+import com.amazonaws.client.builder.AwsClientBuilder.EndpointConfiguration
+import com.amazonaws.services.dynamodbv2.model.BatchWriteItemResult
+
+// Logging
+import org.slf4j.LoggerFactory
+
+// Scala
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+// Generated
+import com.snowplowanalytics.piinguin.server.generated.protocols.piinguin._
+
+/**
+ * Piinguin Client for dynamo
+ * @param client the amazon dynamo client to be used
+ * @param tableName the name of the table to be used (it must exist)
+ */
+class DynamoDBClient(val client: AmazonDynamoDBAsync, val tableName: String)(implicit ec: ExecutionContext) {
+  private val logger                  = LoggerFactory.getLogger(classOf[DynamoDBClient].getName)
+  private val table: Table[PiiRecord] = Table[PiiRecord](tableName)
+
+  /**
+   * Write a single record asynchronously
+   * @param record PiiRecord object to be written
+   * @return a future for the execution of the request with either an error or the odl record if replaced
+   */
+  def putRecord(record: PiiRecord): Future[Either[String, String]] = exec(table.put(record)).transform {
+    case Success(None)                   => ok("OK")
+    case Success(Some(Right(piiRecord))) => ok(s"Replaces $piiRecord")
+    case Success(Some(Left(error)))      => err(s"$error")
+    case Failure(f)                      => err(s"${f.getMessage}")
+  }
+
+  /**
+   * Get a single record asynchronously
+   * @param modifiedValue the value for the table key
+   * @return a future for the execution of the request with either an error or the record if found
+   */
+  def getRecord(modifiedValue: String): Future[Either[String, PiiRecord]] =
+    exec(table.get('modifiedValue -> modifiedValue)).transform {
+      case Success(None) => {
+        logger.debug("Reading record: not found")
```

Comment:
Let's maybe mention what record was not found: `s"Reading record: $modifiedValue not found"`

Reply:
this hasn't been solved afaict